lib/middlewares/src/metering.rs

```diff
@@ -39,43 +43,51 @@ pub struct FunctionMetering<F: Fn(&Operator) -> u64 + Copy + Clone + Send + Sync
     /// The global index in the current module for remaining points.
     remaining_points_index: GlobalIndex,
 
+    error_code_index: GlobalIndex,
+
     /// Accumulated cost of the current basic block.
     accumulated_cost: u64,
 }
 
+pub enum MeteringError {
+    NoError,
+    OutOfGas,
+}
+
+impl <T>TryFrom<Value<T>> for MeteringError {
+    type Error = ();
+
+    fn try_from(v: Value<T>) -> Result<Self, Self::Error> {
+        match v {
+            Value::I32(value) => {
+                match value {
+                    value if value == MeteringError::NoError as _ => Ok(MeteringError::NoError),
+                    value if value == MeteringError::OutOfGas as _ => Ok(MeteringError::OutOfGas),
+                    _ => Err(()),
+                }
+            },
+            Value::I64(value ) => {
+                match value  {
+                    value  if value  == MeteringError::NoError as i64 => Ok(MeteringError::NoError),
+                    value  if value  == MeteringError::OutOfGas as i64 => Ok(MeteringError::OutOfGas),
+                    _ => Err(()),
+                }
+            }
```

Comment:
Why support two different Wasm `Value` types for this conversion? We know in which type the error code is stored, right?

Reply:
indeed, TBH I don't know why I covered both i32 and i64 🤔